### PR TITLE
Recommend making create spans disableable

### LINKED ISSUE #1273: Messaging batch-publish: make per-message tracing disableable
Based on the discussion in https://github.com/open-telemetry/opentelemetry-specification/issues/4123, we should recommend instrumentation libraries to let users control per-message tracing:

- it should be possible to opt-out of per-message tracing (via sampling or tracer suppression)
- if user opts-out, we can fallback to injecting ambient context or publish span context - this should be specified.
- maybe we can call out that for single-publish scenarios per-message span SHOULD NOT be created

Currently spec leaves  it up to the instrumentation library to decide whether to trace messages independently or not.

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #1273

## Changes

Provides guidance on when to create "Create" spans (when instrumenting batch-send) and recommends allowing to disable "Create" spans for batching operations.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Review by trask — APPROVED** [review]

File: docs/messaging/messaging-spans.md [hunk 1]
```diff
@@ -238,6 +238,14 @@ into the message.
 The "Publish" span SHOULD always link to the creation context that was injected
 into a message either from a "Create" span or as a custom creation context.
 
+When instrumenting library API that always sends a single message, it's
+RECOMMENDED to create "Publish" span without "Create" span.
+
+When instrumenting library API that usually operate with batches, it's
+RECOMMENDED to create "Create" span for each message along with the "Publish" span.
+It's also RECOMMENDED to provide a configuration option allowing to disable "Create"
```
Comment (lmolkova) [inline docs/messaging/messaging-spans.md:246]: Let's add an example for no-create-batch-publish

**Review by carlosalberto — APPROVED** [review]

File: docs/messaging/messaging-spans.md [hunk 2]
```diff
@@ -238,6 +239,14 @@ into the message.
 The "Publish" span SHOULD always link to the creation context that was injected
 into a message either from a "Create" span or as a custom creation context.
 
+When instrumenting library API that always sends a single message, it's
```
Comment (joaopgrassi) [inline docs/messaging/messaging-spans.md:242]:
```suggestion
When instrumenting a library API that always sends a single message, it is
```

File: docs/messaging/messaging-spans.md [hunk 3]
```diff
@@ -238,6 +239,14 @@ into the message.
...
+RECOMMENDED to create "Publish" span without "Create" span.
+
+When instrumenting library API that usually operate with batches, it's
```
Comment (joaopgrassi) [inline docs/messaging/messaging-spans.md:245]:
```suggestion
When instrumenting a library API that usually operate with batches, it is
```

File: docs/messaging/messaging-spans.md [hunk 4]
```diff
@@ -238,6 +239,14 @@ into the message.
...
+RECOMMENDED to create "Create" span for each message along with the "Publish" span.
```
Comment (joaopgrassi) [inline docs/messaging/messaging-spans.md:246]:
```suggestion
RECOMMENDED to create a "Create" span for each message along with the "Publish" span.
```

File: docs/messaging/messaging-spans.md [hunk 5]
```diff
@@ -238,6 +239,14 @@ into the message.
...
+It's also RECOMMENDED to provide a configuration option allowing to disable "Create"
```
Comment (joaopgrassi) [inline docs/messaging/messaging-spans.md:247]:
```suggestion
It is also RECOMMENDED to provide a configuration option allowing to disable "Create"
```

**Review by joaopgrassi — APPROVED** [review]

**Pull request merged**